plugins/nexus-repository-manager/src/components/ArtifactTable/ArtifactTable.test.tsx

```diff
@@ -0,0 +1,254 @@
+import React from 'react';
+
+import { render, screen, waitFor } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+
+import { ArtifactRowData, ArtifactTable } from './ArtifactTable';
+
+describe('ArtifactTable', () => {
+  const sampleRows: ArtifactRowData[] = [
+    {
+      version: '1.0.0',
+      artifact: 'test-artifact',
+      assetVariants: new Set(['test-variant']),
+      repositoryType: 'test-repo',
+      hash: {
+        algorithm: 'sha256',
+        value: 'sha256-test-hash',
+      },
+      lastModified: 'Sep 26, 2023, 10:35 PM',
+      sizeBytes: 140000,
+    },
+    {
+      version: '1.0.1',
+      artifact: 'test-artifact',
+      assetVariants: new Set(),
+      repositoryType: 'test-repo',
+      hash: {
+        algorithm: 'sha1',
+        value: 'test-hash-sha1',
+      },
+      lastModified: 'Sep 29, 2023, 10:35 PM',
+      sizeBytes: 130000,
+    },
+  ];
+
+  // Get the text at row[columnName]
+  function getCellText(table: HTMLElement, row: number, columnName: string) {
+    const header = table.querySelector(`thead tr`);
+    if (!header) {
+      throw new Error('No header found');
+    }
+    const columnIndex = Array.from(header.children).findIndex(
+      el => el.textContent === columnName,
+    );
+    if (columnIndex === -1) {
+      throw new Error(`Column ${columnName} not found`);
+    }
+
+    const rowElement = table.querySelector(`tbody tr:nth-child(${row + 1})`);
+    if (!rowElement) {
+      throw new Error(`Row ${row} not found`);
+    }
+    const cell = rowElement.children[columnIndex];
+    return cell.textContent;
+  }
```

Comment:
Should this function be moved outside of the `describe` block?

Reply:
Guessing by your question the answer is yes, so I moved it 😅 

I kept it inside the describe block because that is the narrowest scope possible; but I can see how the whole test file is basically the same, given there aren't any other `describe`s. Can I ask what the heuristic is? Should `sampleRows` be moved as well?